src/mlpack/core/tree/spill_tree/spill_tree_impl.hpp

```diff
@@ -0,0 +1,761 @@
+  arma::vec center, leftCenter, rightCenter;
+  Center(center);
+  left->Center(leftCenter);
+  right->Center(rightCenter);
+
+  const ElemType leftParentDistance = MetricType::Evaluate(center, leftCenter);
+  const ElemType rightParentDistance = MetricType::Evaluate(center,
+      rightCenter);
+
+  left->ParentDistance() = leftParentDistance;
+  right->ParentDistance() = rightParentDistance;
+}
+
+template<typename MetricType,
+         typename StatisticType,
+         typename MatType,
+         template<typename HyperplaneMetricType> class HyperplaneType,
+         template<typename SplitMetricType, typename SplitMatType>
+             class SplitType>
+bool SpillTree<MetricType, StatisticType, MatType, HyperplaneType, SplitType>::
+    SplitPoints(const double tau,
+                const double rho,
+                const arma::Col<size_t>& points,
+                arma::Col<size_t>& leftPoints,
+                arma::Col<size_t>& rightPoints)
+{
+  arma::vec projections(points.n_elem);
+  size_t left = 0, right = 0, leftFrontier = 0, rightFrontier = 0;
+
+  // Count the number of points to the left/right of the splitting hyperplane.
+  for (size_t i = 0; i < points.n_elem; i++)
+  {
+    // Store projection value for future use.
+    projections[i] = hyperplane.Project(dataset->col(points[i]));
+    if (projections[i] <= 0)
+    {
+      left++;
+      if (projections[i] > -tau)
+        leftFrontier++;
+    }
+    else
+    {
+      right++;
+      if (projections[i] < tau)
+        rightFrontier++;
+    }
+  }
+
+  const double p1 = double (left + rightFrontier) / points.n_elem;
+  const double p2 = double (right + leftFrontier) / points.n_elem;
+
+  if ((p1 <= rho || rightFrontier == 0) &&
+      (p2 <= rho || leftFrontier == 0))
+  {
+    // Perform the actual splitting considering the overlapping buffer.  Points
+    // with projection value in the range (-tau, tau) are included in both,
+    // leftPoints and rightPoints.
+    leftPoints.resize(left + rightFrontier);
+    rightPoints.resize(right + leftFrontier);
+    for (size_t i = 0, rc = 0, lc = 0; i < points.n_elem; i++)
+    {
+      if (projections[i] < tau || projections[i] <= 0)
+      {
+        leftPoints[lc] = points[i];
+        lc++;
+      }
+      if (projections[i] > -tau)
+      {
+        rightPoints[rc] = points[i];
+        rc++;
+      }
+    }
+    // Return true, because it is a overlapping node.
+    return true;
+  }
+
+  // Perform the actual splitting ignoring the overlapping buffer.  Points
+  // with projection value less than or equal to zero are included in leftPoints
+  // and points with projection value greater than zero are included in
+  // rightPoints.
+  leftPoints.resize(left);
+  rightPoints.resize(right);
+  for (size_t i = 0, rc = 0, lc = 0; i < points.n_elem; i++)
+  {
+    if (projections[i] <= 0)
+    {
+      leftPoints[lc] = points[i];
+      lc++;
+    }
+    else
+    {
+      rightPoints[rc] = points[i];
+      rc++;
+    }
```

Comment:
Yes, I agree. Done in: https://github.com/mlpack/mlpack/pull/747/commits/dce67ec82adedd7be4c113eea7dec95385f755aa